Pulumi release notes 103 (#4163)
pulumi/pulumi-hugo: themes/default/content/blog/pulumi-release-notes-103/index.md

---

> title: "Pulumi Release Notes: Pulumi ESC, Pulumi Dev Releases, and more!"

Reviewer: if this is too long you can add allow_long_title: True

allow_long_title: true
date: 2024-04-12T11:06:04-08:00
draft: false

> meta_desc: The latest Pulumi updates include Pulumi ESC, Pulumi AI, Pulumi CLI & SDK Dev Releases, and new Pulumi providers

Reviewer: ```suggestion
meta_desc: The latest Pulumi updates include a new Pulumi ESC Key Value Editor, Pulumi CLI & SDK Dev Releases, and more!
```

Pulumi ESC now offers [subject customization](/docs/pulumi-cloud/oidc/azure/#subject-customization) for OIDC tokens, enhancing security across AWS, GCP and Azure by aligning federated credentials with specific identifiers. By configuring the `subjectAttributes` setting, users can include specific environment, user, and organization information in the [OIDC](/docs/pulumi-cloud/oidc/) token subject claim, offering more granular control and customization for cloud resource access. This capability is especially valuable for customers using Azure, where subject claims are required to precisely match the string specified in the policy.

### Pulumi ESC Audit Logs

Pulumi ESC now features comprehensive Audit Logs for all Create, Read, Open, Update, and Delete (CRUD) operations, strengthening your cloud security framework. This critical addition to Pulumi ESC enables organizations to closely monitor and audit their cloud infrastructure changes, ensuring enhanced oversight and security compliance. Through detailed logging, teams gain valuable insights into their operational activities, facilitating improved governance and accountability across cloud resources.

## Core

> ### Dev Releases for the Pulumi CLI and SDKs

Reviewer 2: While we're summarizing other blog posts, can we also add another one for Node.js Monorepos? Basically summarizing and linking to https://www.pulumi.com/blog/nx-monorepo/

Author: Usually, we add only 'new' releases to this blog. Did we do anything to support node monorepos, or it is more of a 'how-to' guide?

Reviewer 2: > Did we do anything to support node monorepos

Yes, we made several product enhancements to enable this (e.g. https://github.com/pulumi/pulumi/issues/12688, https://github.com/pulumi/pulumi/issues/8865)

Author: Is this summary good? Anything you'd add? 

Title: Node.js Monorepo Support 
Summary: Pulumi now seamlessly integrates with Node.js monorepos using tools like Nx. We've enhanced Pulumi to support yarn/npm workspaces, widely used in Node.js monorepos. This enhancement supports npm and Yarn workspaces within monorepos, facilitating better project management by automatically handling dependencies and build processes. This integration allows Pulumi infrastructure codes to coexist with application codes in the same repository, improving development workflow efficiency by using Nx's capabilities to manage build and deploy dependencies effectively.  Check out the [blog post](/blog/nx-monorepo/) for more details.

Reviewer 2: This looks good to me

We introduced [Dev Releases](/blog/introducing-dev-releases/), allowing developers to access the latest features and improvements in Pulumi's infrastructure as code tools ahead of their official release. With this initiative, we aim to engage our community and gather early feedback, enhancing the product development process. This is crucial for identifying and resolving potential issues early, contributing to a smoother transition to new releases. Dev Releases will also be at the cutting edge of Pulumi's evolving capabilities, with the latest bug fixes and features available within minutes of Pulumi developers merging. For instructions on how to use the CLI Dev releases see the [Install Docs](/docs/install/#installing-dev-releases), and for pre-release versions of Dev SDKs see the docs for [Node.js](/docs/languages-sdks/javascript/#dev-versions), [Python](/docs/languages-sdks/python/#dev-versions), [.NET](docs/languages-sdks/dotnet/#dev-versions), and [Go](/docs/languages-sdks/go/#dev-versions).

### Automation API support to install Pulumi CLI 

> The Pulumi Automation API now offers the functionality to [install the Pulumi CLI](https://pkg.go.dev/github.com/pulumi/pulumi/sdk/v3/go/auto#InstallPulumiCommand) directly through the API. This feature enables deploying programs utilizing the Automation API without the need for prior CLI installation. Explore the [example code](https://github.com/pulumi/automation-api-examples/tree/main/go/cli_installation) to see how to integrate the new API seamlessly. 

Reviewer 2: It's unfortunate we only have an example in Go. This is usable from the other languages that support automation API. Perhaps worth at least linking to the API docs for some of the other languages: 
- Node.js: https://www.pulumi.com/docs/reference/pkg/nodejs/pulumi/pulumi/classes/automation.PulumiCommand.html#install
- Python: https://www.pulumi.com/docs/reference/pkg/python/pulumi/#pulumi.automation.PulumiCommand.install
- .NET: Unfortunately, we need to manually kick off a doc update for the new API to show up in https://www.pulumi.com/docs/reference/pkg/dotnet/Pulumi.Automation/Pulumi.Automation.html. In the meantime, we could link directly to https://github.com/pulumi/pulumi-dotnet/blob/edf4a5e904110f198d579d8db0b178aa9af9cdfa/sdk/Pulumi.Automation/Commands/LocalPulumiCommand.cs#L107

Author: I added them

> The Pulumi Automation API now offers the functionality to [install the Pulumi CLI](https://pkg.go.dev/github.com/pulumi/pulumi/sdk/v3/go/auto#InstallPulumiCommand) directly through the API. This feature enables deploying programs utilizing the Automation API without the need for prior CLI installation. Explore the example code in [go](https://github.com/pulumi/automation-api-examples/tree/main/go/cli_installation), [node.js](docs/reference/pkg/nodejs/pulumi/pulumi/classes/automation.PulumiCommand.html#install), [python](/docs/reference/pkg/python/pulumi/#pulumi.automation.PulumiCommand.install), and [.NET](https://github.com/pulumi/pulumi-dotnet/blob/edf4a5e904110f198d579d8db0b178aa9af9cdfa/sdk/Pulumi.Automation/Commands/LocalPulumiCommand.cs#L107) to see how to integrate the new API seamlessly. 

Reviewer 2: The docs links are listed among the Go example. How about re-arranging so we call out just the Go example, and then link to API docs for all the languages? Something like

```suggestion
The Pulumi Automation API now offers the functionality to install the Pulumi CLI directly through the API. This feature enables deploying programs utilizing the Automation API without the need for prior CLI installation. Explore an [example in Go](https://github.com/pulumi/automation-api-examples/tree/main/go/cli_installation) and refer to the new APIs in [Go](https://pkg.go.dev/github.com/pulumi/pulumi/sdk/v3/go/auto#InstallPulumiCommand), [Node.js](docs/reference/pkg/nodejs/pulumi/pulumi/classes/automation.PulumiCommand.html#install), [Python](/docs/reference/pkg/python/pulumi/#pulumi.automation.PulumiCommand.install), and [.NET](https://github.com/pulumi/pulumi-dotnet/blob/edf4a5e904110f198d579d8db0b178aa9af9cdfa/sdk/Pulumi.Automation/Commands/LocalPulumiCommand.cs#L107).
```

---

> title: "Pulumi Release Notes: Pulumi ESC Table Editor, 1Password Support, Pulumi Dev Releases, and more!"

Reviewer 3: ```suggestion
title: "Pulumi Release Notes: Pulumi Dev Releases, Pulumi ESC Table Editor, 1Password Support, and more!"
```

<!--more-->

- [AI](#ai)
  - [Pulumi AI Evolution](#pulumi-ai-evolution)
- [Pulumi Cloud](#pulumi-cloud)
  - [Pulumi ESC 1Password Support](#pulumi-esc-1password-support)
  - [Pulumi ESC Key-Value Table Editor](#pulumi-esc-key-value-table-editor)
  - [Pulumi ESC YAML Code Editor Enhancements](#pulumi-esc-yaml-code-editor-enhancements)
  - [Pulumi ESC OIDC customization](#pulumi-esc-oidc-customization)
  - [Pulumi ESC Audit Logs](#pulumi-esc-audit-logs)
- [Core](#core)
  - [Dev Releases for the Pulumi CLI and SDKs](#dev-releases-for-the-pulumi-cli-and-sdks)
  - [Automation API support to install Pulumi CLI](#automation-api-support-to-install-pulumi-cli)
  - [Node.js Monorepo Support](#nodejs-monorepo-support)
- [Providers and Packages](#providers-and-packages)
  - [New Provider Resources](#new-provider-resources)
  - [New Community Providers](#new-community-providers)
  - [Pulumi Registry Documentation Update](#pulumi-registry-documentation-update)
- [Wrap up](#wrap-up)

## AI

> ### Pulumi AI Evolution

Reviewer 3: Might move this to lower down the list.
